src/POData/ObjectModel/ModelDeserialiser.php

```diff
@@ -50,7 +50,8 @@ public function bulkDeserialise(ResourceEntityType $entityType, ODataEntry $payl
             foreach ($rawProp as $prop) {
                 $propName = $prop->getName();
                 if (!in_array($propName, $keyNames) && !($prop->getResourceType() instanceof ResourceEntityType)) {
-                    $nonRelProp[$propName] = $prop;
+                    $nonRelProp[] = $propName;
+                    $nonRelProp[] = strtolower($propName);
```

Comment:
Note both raw property name and lowercased version thereof, as a fallback against case mismatches